pkg/bloomcompactor/router.go

```diff
@@ -0,0 +1,224 @@
+package bloomcompactor
+
+import (
+	"context"
+	"fmt"
+	"math"
+	"sync"
+	"time"
+
+	"github.com/go-kit/log"
+	"github.com/go-kit/log/level"
+	"github.com/grafana/dskit/concurrency"
+	"github.com/grafana/dskit/multierror"
+	"github.com/pkg/errors"
+	"github.com/prometheus/common/model"
+
+	v1 "github.com/grafana/loki/pkg/storage/bloom/v1"
+	"github.com/grafana/loki/pkg/storage/config"
+)
+
+type DayTable model.Time
+
+func (d DayTable) String() string {
+	return fmt.Sprintf("%d", d.ModelTime().Time().UnixNano()/int64(config.ObjectStorageIndexRequiredPeriod))
```

Comment:
At some point we need to add the prefix from the schema config.

Reply:
A string as table representation is quite confusing, because it may or may not contain the prefix.